x-pack/plugins/security_solution/cypress/e2e/detection_alerts/alerts_charts.cy.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getNewRule } from '../../objects/rule';
+import { ALERTS_HISTOGRAM_LEGEND } from '../../screens/alerts';
+import { waitForAlerts, selectAlertsHistogram } from '../../tasks/alerts';
+import { createCustomRuleEnabled } from '../../tasks/api_calls/rules';
+import { cleanKibana } from '../../tasks/common';
+import { login, visit } from '../../tasks/login';
+import { ALERTS_URL } from '../../urls/navigation';
+import { GLOBAL_SEARCH_BAR_FILTER_ITEM } from '../../screens/search_bar';
+import { HOVER_ACTIONS, TIMELINE_DATA_PROVIDERS_CONTAINER } from '../../screens/timeline';
+import { closeTimelineUsingCloseButton } from '../../tasks/security_main';
+import { openActiveTimeline } from '../../tasks/timeline';
+
+describe('Alerts page charts', () => {
+  before(() => {
+    cleanKibana();
+    login();
+    createCustomRuleEnabled(getNewRule(), 'new custom rule');
+  });
+
+  context('Histogram legend hover actions', () => {
+    beforeEach(() => {
```

Comment:
actually, looking at Jatin's [PR](https://github.com/elastic/kibana/pull/151027), you probably also don't need to do this step before each test. I feel like you could move this 3 lines to the before section above and run the 3 tests without refreshing the UI... WDYT?

Reply:
agree! Here is what i proposed: the outer `before` should set up the alerts page, the `before` under `Histogram` should just click the trend button to make sure we are landing in the right section. The reason I separate them out is because I want to set up a framework to include other chart options (treemap, etc.). but that will come from a separate PR
```
describe('Alerts page charts', () => {
  before(() => {
    cleanKibana();
    login();
    createCustomRuleEnabled(getNewRule(), 'new custom rule');
    visit(ALERTS_URL);
    waitForAlerts();
  });

  describe('Histogram legend hover actions', () => {
    before(() => {
      selectAlertsHistogram();
    });
   ...
```